app/controllers/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user, except: %i[login new create]
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:current_user_id] = @user.id
+      redirect_to root_path, notice: 'Logged in!'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.includes(:created_events).find(params[:id])
+    current_user = @user.attended_events
+    @past_events = current_user.previous_events
+    @upcoming_events = current_user.upcoming_events
```

Comment:
- You are creating a variable called current_user that is not storing the current user but the events they are attending. Remember the variables should have descriptive names and this is breaking that rule and it's also confusing for the people reviewing your code. Please change the name.